lib/resourcemerge/batch.go

```diff
@@ -8,21 +8,26 @@ import (
 // EnsureJob ensures that the existing matches the required.
 // modified is set to true when existing had to be updated with required.
 func EnsureJob(modified *bool, existing *batchv1.Job, required batchv1.Job) {
-	EnsureObjectMeta(modified, &existing.ObjectMeta, required.ObjectMeta)
 
-	if existing.Spec.Selector == nil {
-		*modified = true
-		existing.Spec.Selector = required.Spec.Selector
-	}
-	if !equality.Semantic.DeepEqual(existing.Spec.Selector, required.Spec.Selector) {
-		*modified = true
-		existing.Spec.Selector = required.Spec.Selector
+	// To set Selector you must set ManualSelector to True. If 'required'
+	// sets Selector to value not equal to existing, let current Job
+	// finish as Selector is immutable.
+	if required.Spec.ManualSelector != nil {
+
+		// If ManualSelector is not True we don't care about Selector
+		if *required.Spec.ManualSelector {
+			if !equality.Semantic.DeepEqual(existing.Spec.Selector, required.Spec.Selector) {
+				return
+			}
+		}
+		setBoolPtr(modified, &existing.Spec.ManualSelector, required.Spec.ManualSelector)
 	}
+	EnsureObjectMeta(modified, &existing.ObjectMeta, required.ObjectMeta)
```

Comment:
Why shift this down?  Can we leave it where it was to make the diff smaller and easier to read?

Reply:
Because, if my logic stands, my first check is to see if I need to bail on doing any modifications.

Reply:
> ...my first check is to see if I need to bail on doing any modifications

These in-memory edits are not expensive.  And [we shouldn't be bailing][1] anyway.  So I don't think we need to shift this around in an attempt to save cycles.

[1]: https://github.com/openshift/cluster-version-operator/pull/312/#discussion_r374947620